app/gui/view/debug_scene/cloud-dashboard/src/lib.rs

```diff
@@ -37,131 +37,483 @@
+        SharedWith,
+        Labels,
+        #[display(fmt = "Data Access")]
+        DataAccess,
+        #[display(fmt = "Usage Plan")]
+        UsagePlan,
+    }
+
 
-    ensogl_core::define_shape_system! {
-        (style:Style) {
-            let width = Var::<Pixels>::from("input_size.x");
-            let height = Var::<Pixels>::from("input_size.y");
-            Rect((width, height)).corners_radius(5.0.px()).fill(color::Rgba(0.655, 0.788, 0.624, 1.0)).into()
+    // === Main `impl` ===
+
+    impl Columns {
+        /// Number of columns that we want to display in the grid view. This corresponds to the
+        /// number of variants in this enum, and must be updated whenever the enum is changed.
+        // FIXME [NP]: Should we adjust this manually, use `std::mem::variant_count`, or use a proc
+        // macro?
+        const LEN: usize = 6;
+
+        /// Returns the [`Columns`] variant corresponding to the given discriminant value.
+        ///
+        /// If the discriminant is out of bounds, returns `None`.
+        ///
+        /// [`Columns`]: ::crate::projects_grid::Columns
+        // FIXME [NP]: Are there any standard library functions that we can use instead? Or perhaps
+        // a proc macro?
+        fn from_discriminant(discriminant: usize) -> Option<Self> {
+            match discriminant {
+                0 => Some(Self::Projects),
+                1 => Some(Self::LastModified),
+                2 => Some(Self::SharedWith),
+                3 => Some(Self::Labels),
+                4 => Some(Self::DataAccess),
+                5 => Some(Self::UsagePlan),
+                _ => None,
+            }
         }
     }
-}
 
 
-// ========================
-// === Sample Component ===
-// ========================
 
-mod sample_component {
-    use super::*;
+    // ============
+    // === Grid ===
+    // ============
+
+    /// A variant of the [`component`]'s underlying [`ensogl_grid_view`].
+    ///
+    /// Our component is a grid displaying a list of the user's [`Project`]'s. We use a scrollable
+    /// grid since the user may have more [`Project`]s than can fit on the screen. We use a grid
+    /// with headers since the user needs to know which [`Project`] property each column represents.
+    ///
+    /// [`component`]: ::crate::projects_grid::View
+    /// [`ensogl_grid_view`]: ::ensogl_grid_view
+    /// [`Project`]: ::enso_cloud_view::project::Project
+    pub type Grid = ensogl_grid_view::simple::SimpleScrollableSelectableGridViewWithHeaders;
 
+
+
+    // ===========
     // === FRP ===
+    // ===========
 
     ensogl_core::define_endpoints_2! {
-        Input{
-            /// The component browser is displayed on screen.
-            set_size(Vector2),
+        Input {
+            set_projects(Vec<view::project::Project>),
+            model_for_entry((Row, Col), EntryModel),
         }
-        Output{
-            size(Vector2),
+        Output {
+            model_for_entry_needed((Row, Col)),
```

Comment:
This is ugly, please cjust chat with the grid view implementator about making sructs here. Otherwise the code needs to do .1.0.1.1.1.1..1.0.0.0 etc